### PR TITLE
Migrated containers from hibernate to SQL

### PR DESCRIPTION
**Pull request by hassan-mustafa-baig**
### Proposed Changes
* Data access layer for the container portlet has been migrated from Hibernate to SQL. This has fixed the issue https://github.com/dotcms/core/issues/23454. 

### Checklist
- [X] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
** any additional useful context or info **

### Screenshots
Original             |  Updated
:-------------------------:|:-------------------------:
** original screenshot **  |  ** updated screenshot **

### CODE REVIEW

File: dotCMS/src/main/java/com/dotmarketing/portlets/containers/business/ContainerFactoryImpl.java [hunk 1]
```diff
@@ -67,17 +70,122 @@ public class ContainerFactoryImpl implements ContainerFactory {
...
-		containerCache.remove(container);
+		save(container, UUIDGenerator.generateUuid());
 	}
 
-	@Override
-	public void save(final Container container, final String existingId) throws DotDataException {
-		HibernateUtil.saveWithPrimaryKey(container, existingId);
-		containerCache.remove(container);
+	@WrapInTransaction
+	public void save(final Container container, final String inode) throws DotDataException {
```
Comment (wezell) [inline dotCMS/src/main/java/com/dotmarketing/portlets/containers/business/ContainerFactoryImpl.java:105]: This logic seems wrong.  If you pass in an inode, it should be used in the container.  The `inode` param is never used.

File: dotCMS/src/main/java/com/dotmarketing/portlets/containers/business/ContainerSQL.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    }
+    public static ContainerSQL getInstance() {
+        if (_instance == null) {
+            _instance = new ContainerSQL();
+        }
+
+        return _instance;
+    }
+
+    public static final String FIND_BY_INODE = "select container.*, container_identifier.* from dot_containers container, identifier container_identifier "
```
[the remark this answers is not in the document]
Comment (hassan-mustafa-baig) [reply dotCMS/src/main/java/com/dotmarketing/portlets/containers/business/ContainerSQL.java:19]: for the current scope of the ticket https://github.com/dotcms/core/issues/23454  I haven't seen any usage of identifier but maybe for the other functionality we may use it. I basically replicate this from Templates and I only covered insert and updated operations.

File: dotCMS/src/main/java/com/dotmarketing/portlets/containers/business/ContainerFactoryImpl.java (hunk 1 above)
Comment (hassan-mustafa-baig) [reply dotCMS/src/main/java/com/dotmarketing/portlets/containers/business/ContainerFactoryImpl.java:105]: yeah you are right. It should be removed or used. Let me address this. Thanks for your feedback @wezell

Comment (hassan-mustafa-baig) [reply dotCMS/src/main/java/com/dotmarketing/portlets/containers/business/ContainerFactoryImpl.java]: This feedback has been addressed

**Review by jcastro-dotcms — APPROVED** [review]

**Comment by mbiuki** [comment]
@hassan-mustafa-baig - pls take a look at the Sonar Qube report and make sure there are no issues of type Blocker, Critical, and Major, thanks.

**Review by jdotcms — APPROVED** [review]
It is a good change, I would just add a saveNew method that allows the creation of the container without identifier and generates for you the uuid

**Comment by hassan-mustafa-baig** [comment]
Since there were some issues with the branch therefore I have created another PR https://github.com/dotCMS/core/pull/23710

**Pull request closed without merging**